packages/utilities/src/asAsync.tsx

```diff
@@ -0,0 +1,58 @@
+
+import * as React from 'react';
+
+interface IAsyncState<TProps> {
+  component?: React.ComponentType<TProps> | undefined;
+}
+
+export interface IModuleWithDefault<T> {
+  default: T;
+}
+
+/**
+ * Produces a component which internally resolves asynchronously before first mount.
+ * The component passes all props through to the loaded component.
+ */
+export function asAsync<TProps>(
+  load: () => Promise<{ default: React.ComponentType<TProps>; }>
+): React.ComponentType<TProps>;
+export function asAsync<TProps>(
+  load: () => Promise<React.ComponentType<TProps>>
+): React.ComponentType<TProps>;
```

Comment:
what is going on here, why do you have these 2 export function declarations without bodies? I don't quite understand this syntax.

Reply:
The exports are overloads. If the module you `import` has a default export, you get one overload. Otherwise you get the other.